File: solutions/cpp/raindrops/1/raindrops.cpp
#include "raindrops.h"

namespace raindrops {

std::string convert(int n)
{
    std::string sound = "";
    if (n % 3 == 0)
        sound += "Pling";
    if (n % 5 == 0)
        sound += "Plang";
    if (n % 7 == 0)
        sound += "Plong";
    if (sound.length() == 0)
        sound = std::to_string(n);
    return sound;
}

}  // namespace raindrops

File: solutions/r/raindrops/1/raindrops.R
raindrops <- function(number) {
  sound <- ""
  if (number %% 3 == 0) {
    sound <- paste(sound, "Pling", sep="")
  }
  if (number %% 5 == 0) {
    sound <- paste(sound, "Plang", sep="")
  }
  if (number %% 7 == 0) {
    sound <- paste(sound, "Plong", sep="")
  }
  if (nchar(sound) == 0) {
    sound <- sprintf("%d", number)
  }
  return(sound)
}
